remix/app/routes/_app.newsletter.update.$id.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect } from 'react';
+
+import { ActionArgs, json } from '@remix-run/node';
+import { Form, useActionData, useNavigate } from '@remix-run/react';
+import { withZod } from '@remix-validated-form/with-zod';
+import { validationError } from 'remix-validated-form';
+import { z } from 'zod';
+
+import { addNotification } from '../helpers/localStorage.helper';
+import appHandler from '../lib/handler/app.handler';
+import NewsletterRepository from '../repositories/newsletter.server';
+
+const validator = withZod(
+  z.object({
+    name: z.string().min(1, { message: 'Name is required' }),
+    content: z.string().min(1, { message: 'Content is required' }),
+  })
+);
+
+export async function action({ request, params }: ActionArgs) {
+  return appHandler(request, async (user) => {
+    const result = await validator.validate(await request.formData());
+
+    if (result.error) {
+      return validationError(result.error);
+    }
+
+    const newsletterExists = await NewsletterRepository.findOne({
+      where: { id: params.id },
+    });
+
+    if (!newsletterExists) {
+      return validationError({
+        fieldErrors: {
+          Error: 'Newsletter not found',
+        },
+      });
+    }
+
+    if (newsletterExists.userId !== user.id) {
```

Comment:
Or you can use use [Prisma's findMany](https://www.prisma.io/docs/concepts/components/prisma-client/crud#get-all-records), which will also allow you to pass the `user.id` to the query and we will be able to check both things(the newsletter id exists, and the user of the newsletter is the current user) 👀